adapters/oidc/adapter-core/src/main/java/org/keycloak/adapters/OAuthRequestAuthenticator.java

```diff
@@ -335,28 +336,26 @@ protected AuthChallenge resolveCode(String code) {
             String httpSessionId = deployment.getTokenStore() == TokenStore.SESSION ? reqAuthenticator.changeHttpSessionId(true) : null;
             tokenResponse = ServerRequest.invokeAccessCodeToToken(deployment, code, rewrittenRedirectUri(strippedOauthParametersRequestUri), httpSessionId);
         } catch (ServerRequest.HttpFailure failure) {
-            log.error("failed to turn code into token");
-            log.error("status from server: " + failure.getStatus());
+            log.log(Level.SEVERE,"failed to turn code into token");
+            log.log(Level.SEVERE,"status from server: " + failure.getStatus());
             if (failure.getError() != null) {
-                log.error("   " + failure.getError());
+                log.log(Level.SEVERE,"   " + failure.getError());
             }
             return challenge(403, OIDCAuthenticationError.Reason.CODE_TO_TOKEN_FAILURE, null);
 
         } catch (IOException e) {
-            log.error("failed to turn code into token", e);
+            log.log(Level.SEVERE,"failed to turn code into token", e);
             return challenge(403, OIDCAuthenticationError.Reason.CODE_TO_TOKEN_FAILURE, null);
         }
 
         tokenString = tokenResponse.getToken();
         refreshToken = tokenResponse.getRefreshToken();
         idTokenString = tokenResponse.getIdToken();
 
-        log.debug("Verifying tokens");
-        if (log.isTraceEnabled()) {
-            logToken("\taccess_token", tokenString);
-            logToken("\tid_token", idTokenString);
-            logToken("\trefresh_token", refreshToken);
-        }
+        log.log(Level.FINE, "Verifying tokens");
+        logToken("\taccess_token", tokenString);
```

Comment:
Don't remove the condition here.

Reply:
you are right, in fact, I will put the if in the logToken, since whoever is using this method, shouldn't bother much about logging impact.